src/it/java/org/jboss/tools/intellij/quarkus/utils/ProjectToolWindowUtils.java

```diff
@@ -0,0 +1,75 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.intellij.quarkus.utils;
+
+import com.intellij.remoterobot.RemoteRobot;
+import org.assertj.swing.core.MouseButton;
+import org.jboss.tools.intellij.quarkus.fixtures.mainIdeWindow.ProjectToolWindowFixture;
+import org.jboss.tools.intellij.quarkus.fixtures.mainIdeWindow.ToolWindowsPaneFixture;
+
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+import static com.intellij.remoterobot.stepsProcessing.StepWorkerKt.step;
+import static com.intellij.remoterobot.utils.RepeatUtilsKt.waitFor;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * Static utilities that assist and simplify working with the Project tool window
+ *
+ * @author zcervink@redhat.com
+ */
+public class ProjectToolWindowUtils {
+
+    public static boolean isAProjectFilePresent(RemoteRobot remoteRobot, String... path) {
+        try {
+            navigateThroughTheProjectTree(remoteRobot, ActionToPerform.HIGHLIGHT, path);
+        } catch (NoSuchElementException e) {
+            return false;
+        }
+        return true;
+    }
+
+    private static void navigateThroughTheProjectTree(RemoteRobot remoteRobot, ActionToPerform action, String... pathArray) {
+        step("Navigate through the project tree", () -> {
+            List<String> path = Arrays.asList(pathArray);
+            Iterator<String> pathIterator = path.iterator();
+            while (pathIterator.hasNext()) {
+                final ProjectToolWindowFixture projectToolWindowFixture = remoteRobot.find(ProjectToolWindowFixture.class);
```

Comment:
No, it could not. It needs to be performed in every iteration, because the visible items changes in every iteration. The ProjectViewTree containing the file and folder/package names extends when you expand a folder/package. If it is performed outside the loop, the firstly hidden file/folder items are not accessible.

### See the "src" folder - it contains some other items. The items are hidden before expanding the "src" folder therefore they are not accessible using the RR framework right now:
<img width="343" alt="Snímek obrazovky 2021-02-03 v 0 27 42" src="https://user-images.githubusercontent.com/43820055/106676181-b3354c80-65b6-11eb-952a-2cd9046f3494.png">

Reply:
No, it could not. It needs to be performed in every iteration, because the visible items changes in every iteration. The ProjectViewTree containing the file and folder/package names extends when you expand a folder/package. If it is performed outside the loop, the firstly hidden file/folder items are not accessible.

### See the "src" folder - it contains some other items. The items are hidden before expanding the "src" folder therefore they are not accessible using the RR framework right now:

<img width="343" alt="Snímek obrazovky 2021-02-03 v 0 27 42" src="https://user-images.githubusercontent.com/43820055/106676790-d14f7c80-65b7-11eb-9158-acd2ce79c934.png">

Reply:
UPDATE: It could be moved outside the loop. Fixed.